arrow-core-data/src/main/kotlin/arrow/core/Option.kt

```diff
@@ -614,10 +1043,317 @@ fun <T> Iterable<T>.lastOrNone(predicate: (T) -> Boolean): Option<T> = this.last
+}
+
+/**
+ * Separate the inner [Validated] value into the [Validated.Invalid] and [Validated.Valid].
+ *
+ * @receiver Option of Either
+ * @return a tuple containing Option of [Validated.Invalid] and another Option of its [Validated.Valid] value.
+ */
+fun <A, B> Option<Validated<A, B>>.separateValidated(): Pair<Option<A>, Option<B>> {
+  val asep = flatMap { gab -> gab.fold({ Some(it) }, { None }) }
+  val bsep = flatMap { gab -> gab.fold({ None }, { Some(it) }) }
+  return asep to bsep
+}
+
+fun <A> Option<Iterable<A>>.sequence(): List<Option<A>> =
+  traverse(::identity)
+
+fun <A> Option<Iterable<A>>.sequence_(): List<Unit> =
+  traverse_(::identity)
+
+fun <A, B> Option<Either<A, B>>.sequenceEither(): Either<A, Option<B>> =
+  traverseEither(::identity)
+
+fun <A, B> Option<Either<A, B>>.sequenceEither_(): Either<A, Unit> =
+  traverseEither_(::identity)
+
+fun <A, B> Option<Validated<A, B>>.sequenceValidated(): Validated<A, Option<B>> =
+  traverseValidated(::identity)
+
+fun <A, B> Option<Validated<A, B>>.sequenceValidated_(): Validated<A, Unit> =
+  traverseValidated_(::identity)
+
+fun <A, B> Option<Ior<A, B>>.unalign(): Pair<Option<A>, Option<B>> =
+  unalign(::identity)
+
+inline fun <A, B, C> Option<C>.unalign(f: (C) -> Ior<A, B>): Pair<Option<A>, Option<B>> =
+  when (val option = this.map(f)) {
+    is None -> None to None
+    is Some -> when (val v = option.t) {
+      is Ior.Left -> Some(v.value) to None
+      is Ior.Right -> None to Some(v.value)
+      is Ior.Both -> Some(v.leftValue) to Some(v.rightValue)
+    }
+  }
 
-fun <A, B> Option<A>.zip(fb: Option<B>): Option<Pair<A, B>> =
-  flatMap { a: A -> fb.map { b -> Pair(a, b) } }
+fun <A> Option<Iterable<A>>.unite(MA: Monoid<A>): Option<A> =
+  map { iterable ->
+    iterable.fold(MA)
+  }
+
+fun <A, B> Option<Either<A, B>>.uniteEither(): Option<B> =
+  flatMap { either ->
+    either.fold({ None }, { b -> Some(b) })
+  }
+
+fun <A, B> Option<Validated<A, B>>.uniteValidated(): Option<B> =
+  flatMap { validated ->
+    validated.fold({ None }, { b -> Some(b) })
+  }
+
+fun <A, B> Option<Pair<A, B>>.unzip(): Pair<Option<A>, Option<B>> =
+  unzip(::identity)
+
+inline fun <A, B, C> Option<C>.unzip(f: (C) -> Pair<A, B>): Pair<Option<A>, Option<B>> =
+  fold(
+    { Option.empty<A>() to Option.empty() },
+    { f(it).let { pair -> Some(pair.first) to Some(pair.second) }}
+  )
+
+/**
+ *  Given [A] is a sub type of [B], re-type this value from Option<A> to Option<B>
+ *
+ *  Option<A> -> Option<B>
+ *
+ *  ```kotlin:ank:playground
+ *  import arrow.core.Option
+ *  import arrow.core.some
+ *  import arrow.core.widen
+ *
+ *  fun main(args: Array<String>) {
+ *   val result: Option<CharSequence> =
+ *   //sampleStart
+ *   "Hello".some().map({ "$it World" }).widen()
+ *   //sampleEnd
+ *   println(result)
+ *  }
+ *  ```
+ */
+fun <B, A : B> Option<A>.widen(): Option<B> =
+  this
+
+fun <A> Option<A>.eqv(EQA: Eq<A>, other: Option<A>): Boolean = when (this) {
```

Comment:
These will not be added for the data types. Still need to create a clean-up ticket for the already done types.